open-sphere-base/core/src/main/java/io/opensphere/core/control/keybinding/ControlUI.java

```diff
@@ -0,0 +1,246 @@
+ * to General and Map controls.
+ */
+public class ControlUI extends GridPane
+{
+    /**
+     * The directory of the "plus" icon.
+     */
+    private String myPlusDir = "images/keys/plus.png";
+
+    /**
+     * The directory of the "mouse left button clicked" icon.
+     */
+    private String mouseLeft = "images/keys/MouseLeft.png";
+
+    /**
+     * The directory of the "mouse middle button clicked" icon.
+     */
+    private String mouseMiddle = "images/keys/MouseMiddle.png";
+
+    /**
+     * The directory of the "mouse right button clicked" icon.
+     */
+    private String mouseRight = "images/keys/MouseRight.png";
+
+    /**
+     * @param width the horizontal size desired.
+     * @param height the vertical size desired.
+     */
+    public ControlUI(int width, int height)
+    {
+        setMinSize(width, height);
+        setStyle("-fx-background-color : derive(-fx-base, 18%)");
+        setHgap(10);
+        setVgap(10);
+        setPadding(new Insets(5));
+
+        add(createGeneral(), 0, 0);
+        add(createGeneralMap(), 1, 0);
+        add(createMapZoom(), 0, 1);
+        add(createMapTools(), 1, 1);
+    }
+
+    /**
+     * Creates a sub-window containing Map Tool Controls.
+     * @return mainBox the sub-window contained in a GridPane.
+     */
+    private GridPane createMapTools()
+    {
+        ControlTab thecontrols = new ControlTab();
+        GridPane mainBox = thecontrols.createBlankPane(" Map Tools");
+        GridPane dispArea = thecontrols.getDisplayArea();
+
+        VBox l1 = thecontrols.createLeftLabel("Draw Geometry");
+        HBox r1 = thecontrols.createCustomText("Shift");
+        r1.getChildren().addAll(new ImageView(myPlusDir), thecontrols.customImageView(mouseLeft));
+
+        VBox l2 = thecontrols.createLeftLabel("Toggle Arc Length");
+        HBox r2 = thecontrols.createCustomText("m");
+
+        VBox l3 = thecontrols.createLeftLabel("Live Track Mode");
+        HBox r3 = thecontrols.createCustomText("Ctrl [hold]");
+
+        VBox l4 = thecontrols.createLeftLabel("Bullseye Vector");
+        HBox r4 = thecontrols.createCustomText(",");
+
+        VBox l5 = thecontrols.createLeftLabel("Copy Coordinates");
+        HBox r5 = thecontrols.createCustomText("C");
+
+        VBox l6 = thecontrols.createLeftLabel("Display All Coordinates");
+        HBox r6 = thecontrols.createCustomText(".");
+
+        dispArea.add(l1, 0, 1);
+        dispArea.add(r1, 2, 1);
+        dispArea.add(l2, 0, 2);
+        dispArea.add(r2, 2, 2);
+        dispArea.add(l3, 0, 3);
+        dispArea.add(r3, 2, 3);
+        dispArea.add(l4, 0, 4);
+        dispArea.add(r4, 2, 4);
+        dispArea.add(l5, 0, 5);
+        dispArea.add(r5, 2, 5);
+        dispArea.add(l6, 0, 6);
+        dispArea.add(r6, 2, 6);
+        return mainBox;
+    }
+
+    /**
+     * Creates a sub-window containing the General Controls.
+     * @return mainBox the sub-window contained in a GridPane.
```

Comment:
Leave an empty line between description and return, no need to give the name of the return variable

Reply:
updated